examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt

```diff
@@ -86,6 +87,29 @@ class ReplayHistoryActivity : AppCompatActivity() {
         }
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == HistoryFilesActivity.REQUEST_CODE) {
+            navigationContext?.handleHistoryFileSelected()
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun ReplayNavigationContext.handleHistoryFileSelected() {
+        loadNavigationJob = CoroutineScope(Dispatchers.Main).launch {
+            val events = loadReplayHistory()
+            mapboxReplayer.clearEvents()
+            mapboxReplayer.pushEvents(events)
+            mapboxNavigation.resetTripSession()
```

Comment:
this line is what causes the difference between the videos in the PR description.

The navigator needs to be reset, and then it can start from a new location.